langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingModel.java

```diff
@@ -0,0 +1,78 @@
+package dev.langchain4j.model.bedrock;
+
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.bedrock.internal.AbstractBedrockEmbeddingModel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Bedrock Cohere Embedding Model with support for both versions:
+ * {@code cohere.embed-english-v3} and it's multilingual version {@code cohere.embed-multilingual-v3}
+ * <br>
+ * See more details <a href="https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-embed.html">here</a>.
+ */
+@SuperBuilder
+@Getter
+public class BedrockCohereEmbeddingModel extends AbstractBedrockEmbeddingModel<BedrockCohereEmbeddingResponse> {
+
+    @Builder.Default
+    private final String model = Types.CohereEmbedEnglishTextV3.getValue();
+
+    @Builder.Default
+    private final String inputType = InputType.SEARCH_DOCUMENT.getValue();
+
+    @Override
+    protected List<Map<String, Object>> getRequestParameters(List<TextSegment> textSegments) {
+        List<Map<String, Object>> result = new ArrayList<>();
+        Map<String, Object> parameters = new HashMap<>();
+        for (TextSegment textSegmentItem: textSegments) {
+            parameters.put("texts", new String[] {textSegmentItem.text()});
```

Comment:
Assuming that Cohere supports embedding multiple texts in one request, why not having a single `parameters` map with `texts` containing all texts?

Reply:
Changed according to suggestions